### PR TITLE
Small bug fixes

### PR DESCRIPTION
**Pull request by TomGeorge1234**
Just a couple of small bug fixes:

 * You mixed up student and teacher in the distillation loss which matters in this case since the teacher is detached. 
 * `num_epochs` is an int so has no length. 

(((Brilliant code by the way, thanks for writing and maintaining this!!!)))

### CODE REVIEW

**Comment by mishra-18** [comment]
Hey @TomGeorge1234 
Thnx for your bug fixes!! This means a lot...

I looked at the code again and realized the current implementation is a bit off what's proposed in the paper. I've made the following changes:

- Refactored the global and local crop functions to generate 2 and 6 crops per Image batch.
- Updated the loss function to calculate the loss for now concatnated student and teacher outputs.
- I've also observed the official dino implementation uses cross-view loss estimation (probably to avoid same-view comparisons)

You can take a look at the changes and see if it makes sense.

**Comment by TomGeorge1234** [comment]
Cool, lgtm! I was also going to flag the fact that `local_augment()` had an unused `num_crops` kwarg but it seems like you spotted that too so thanks for incorporating. Great stuff! I'd say it's good to merge...
